actor/src/main/scala/org/apache/pekko/io/dns/internal/DnsClient.scala

```diff
@@ -92,23 +101,26 @@ import pekko.pattern.{ BackoffOpts, BackoffSupervisor }
     case Question4(id, name) =>
       log.debug("Resolving [{}] (A)", name)
       val msg = message(name, id, RecordType.A)
-      inflightRequests += (id -> (sender() -> msg))
-      log.debug("Message [{}] to [{}]: [{}]", id, ns, msg)
-      socket ! Udp.Send(msg.write(), ns)
+      newInflightRequests(msg, sender()) {
```

Comment:
All questions are now protected with `newInflightRequests`. This ensures that transaction id is not already in use and the returns the result.